src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -69,7 +69,19 @@ public function __construct(
         } elseif (!\is_array($data)) {
             throw new \TypeError(sprintf('"%s": Argument $data is expected to be a string or array, got "%s".', __METHOD__, get_debug_type($data)));
         } elseif ([] !== $data) {
```

Comment:
@derrabus shouldn't the `$data` argument itself also be deprecated? Otherwise we cannot really remove the argument from the constructor. So $data should be removed with the appropriate BC layer in place.

Reply:
We don't really have a strategy for phasing out named arguments, do we. I would consider the `$data` parameter deprecated already and my plan is to remove it from the constructor in Symfony 6.

I think we should about how proper deprecation layers for named arguments can look like in a dedicated RFC. And maybe we will fine-tune this deprecation layer for 5.4 then.

Reply:
I'm also not 100% happy with the current deprecation layer, I must admit.